packages/standalone-analyzer/src/index.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  normalizeOptions,
+  Output,
+  type Options,
+  type NormalizedOptions,
+} from "@codecov/bundler-plugin-core";
+import { PLUGIN_NAME, PLUGIN_VERSION } from "./version";
+import { normalizeStandaloneOptions, type StandaloneOptions } from "./options";
+import { getAssets } from "./assets";
+
+/**
+ * Generates a Codecov bundle stats report and optionally uploads it to Codecov. This function can
+ * be imported into your code or used via the standalone-analyzer CLI.
+ *
+ * @param {string} buildDirectoryPath - The path to the build directory containing the production
+ *        assets for the report. Can be absolute or relative.
+ * @param {Options} coreOptions - Configuration options for generating and uploading the report.
+ * @param {StandaloneOptions} [standaloneOptions] - Optional configuration for standalone usage.
+ *
+ * @returns {Promise<void>} A promise that resolves when the report is generated and handled
+ *          (dry-runned or uploaded).
+ *
+ * @example
+ * const buildDir = '/path/to/build/directory'; // absolute or relative path
+ * const coreOpts = {
+ *   dryRun: true,
+ *   uploadToken: 'your-upload-token',
+ *   retryCount: 3,
+ *   apiUrl: 'https://api.codecov.io',
+ *   bundleName: 'my-bundle', // bundle identifier in Codecov
+ *   enableBundleAnalysis: true,
+ *   debug: true,
+ * };
+ * const standaloneOpts = {
+ *   dryRunner: async (report) => console.info('Dry run output: ', report.bundleStatsToJson()),
+ *   reportOverrider: async (original) => original,
+ * };
+ *
+ * CreateAndHandleReport(buildDir, coreOpts, standaloneOpts)
+ *   .then(() => console.log('Report successfully generated and handled.'))
+ *   .catch((error) => console.error('Failed to generate or upload report:', error));
+ */
+export const CreateAndHandleReport = async (
+  buildDirectoryPath: string,
+  coreOptions: Options,
+  standaloneOptions?: StandaloneOptions,
+): Promise<void> => {
+  // normalize options
+  const standaloneOpts = normalizeStandaloneOptions(standaloneOptions);
+  const coreOpts = normalizeOptions(coreOptions);
+  if (!coreOpts.success) {
+    throw new Error("Invalid options: " + coreOpts.errors.join(" "));
+  }
+
+  // create report
+  const initialReport: Output = await makeReport(
+    buildDirectoryPath,
+    coreOpts.options,
+  );
+
+  // override report as needed (by default returns unchanged)
+  const finalReport = await standaloneOpts.reportOverrider(initialReport);
```

Comment:
we need to handle errors here.

Reply:
Ideally we can log something out to indicate that this is an error from the reportOverrider, and then rethrow the error so they get the full stacktrace.

We just want to communicate where exactly the fault lies.

Reply:
Got it - that's a good idea, thanks for the suggestion! I'll grab that in the [follow-up PR](https://github.com/codecov/codecov-javascript-bundler-plugins/pull/163)!